modules/library/main/src/main/java/org/geotools/filter/function/AbstractQuantityClassificationFunction.java

```diff
@@ -0,0 +1,144 @@
+package org.geotools.filter.function;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.util.NullProgressListener;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.visitor.CalcResult;
+import org.geotools.feature.visitor.FeatureCalc;
+import org.opengis.filter.capability.FunctionName;
+
+public abstract class AbstractQuantityClassificationFunction extends ClassificationFunction {
+
+    public AbstractQuantityClassificationFunction(FunctionName name) {
+        super(name);
+    }
+
+    /**
+     * Returns the list visitor for the specific implementation of "quantity" (e.g., count, area,
+     * ...). The visitor must return a "bins" structure matching {code}List<Comparable>[]{code},
+     * where each array entry is a bin, and values inside the bin are sorted from lowest to highest
+     *
+     * @return
+     */
+    protected abstract FeatureCalc getListVisitor();
+
+    private Object calculate(SimpleFeatureCollection featureCollection) {
+        FeatureCalc quantileVisit = getListVisitor();
```

Comment:
Maybe you mean quantityVisit here?

Reply:
Yes, leftover from the extraction of a common superclass from the old Quantile function